source/euclid.js

```diff
@@ -0,0 +1,11 @@
+'use strict';
+
+const gcd = (x, y) => y ? gcd(y, x % y) : x
+
+const euclid = (...numbers) => {
+	if (numbers.some((number) => !Number.isInteger(number))) {
+		return null;
+	}		
+	
+	return numbers.reduce((previousValue, current) => gcd(previousValue, Math.abs(current)), 0);
```

Comment:
А - я понял кажется что тебе не нравилось - такая длинная неудобная конструкция

Но сейчас будет фокус

```suggestion
	return numbers.map(Math.abs).reduce(gcd, 0);
```

https://youtu.be/mZIE-K0z1ZM?si=oUYdREnAbl6yLGFT =)

Reply:
так, с помощью map мы создаем новый массив, где на каждый элемент применили Math.abs (т.е. преобразовали изначальный массив по модулю). и на этот новый полученный массив накинули reduce. 
но вот я не понял, из-за чего пропали аргументы, которые передаются в gcd.
он же не просто на автомате закидывает первые 2 аргумента в нашу функцию?

Reply:
> он же не просто на автомате закидывает первые 2 аргумента в нашу функцию?

=)

просто) в том же самом порядке - это же та же самая функция что ты создавал до этого - первый проп - акумилирующееся значение, второй - очередной элемент массива, третий - индекс массива и четвертый - сам массив

gcd - это функция - у неё первый параметр это x, второй y - остальные ей не интересны

ты например можешь вызвать по этому 
```js
gcd(6, 8, 'biba', {boba: 1})
```
и тебе вернет 2

Reply:
да, я проверил. в старом виде(без map) убрал обозначения аргументов. работало правильно для всего, кроме отрицательных чисел, как и ожидалось